src/core/Akka.Remote/Deadline.cs

```diff
@@ -115,6 +120,16 @@ public static Deadline Now
                 return deadline;
         }
 
+        public static bool operator ==(Deadline left, Deadline right)
+        {
+            return left.Equals(right);
```

Comment:
I just checked the `Equals()` method (line 62) and it is doing a dangerous explicit cast, which will throw if anything else is passed into it. We might as well fix that in this PR.
A ReferenceEquals() shortcut in there would also be good.

Reply:
I don't see an explicit cast, only this:

```csharp
public override bool Equals(object obj) =>
                obj is Deadline deadline && Equals(deadline);
```

Reply:
Hmmm... weird... I'm talking about these code:
https://github.com/akkadotnet/akka.net/blob/aca6850eb68bbdb64f029bad019d3e58e4bb290f/src/core/Akka.Remote/Deadline.cs#L60-L69

Reply:
Found it - very odd